### PR TITLE
LPS-142861 and LPS-142943 Using API to download invitation keys

### PR DESCRIPTION
**Pull request by iagochaves**

### CODE REVIEW

**Comment by jairmedeiros** [comment]
Hi @iagochaves, both 744db8b and 45244ce commits must have messages that indicate only one LPS id.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js [hunk 1]
```diff
@@ -13,11 +15,13 @@ import reducer, {actionTypes} from './reducer';
 const AppContext = createContext();
 
 const AppContextProvider = ({assetsPath, children, page}) => {
+	const {oktaSessionURL} = useApplicationProvider();
 	const dispatchEvent = useCustomEvent(CUSTOM_EVENTS.USER_ACCOUNT);
 	const [state, dispatch] = useReducer(reducer, {
 		assetsPath,
 		page,
 		project: {},
+		sessionId: 'sessionId',
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js:24]:
```suggestion
		sessionId: '',
```

Comment (iagochaves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js:24]: If we remove the mock, the pages won't work until the Okta is finished, so I thought that using that for a while.

**Comment by iagochaves** [comment]
Hello @jairmedeiros, commits renamed!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js [hunk 2]
```diff
@@ -28,11 +32,33 @@ const AppContextProvider = ({assetsPath, children, page}) => {
...
+
+		getSessionId();
+	}, [oktaSessionURL]);
+
 	useEffect(() => {
 		const projectExternalReferenceCode = SearchParams.get(
 			PARAMS_KEYS.PROJECT_APPLICATION_EXTERNAL_REFERENCE_CODE
 		);
 
+		window.addEventListener('customer-portal-menu-selected', ({detail}) => {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js:55]: This event listener should be removed, in the call back of useEffect

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/index.js [hunk 3]
```diff
@@ -29,8 +29,13 @@ const Pages = () => {
 	}
 
 	if (page === pages.COMMERCE) {
-		if (userAccount) {
-			return <ActivationKeys.Commerce accountKey={project.accountKey} />;
+		if (project && sessionId) {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/index.js:32]: Put everything in one if

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/index.js [hunk 4]
```diff
@@ -29,8 +29,13 @@ const Pages = () => {
 	}
 
 	if (page === pages.COMMERCE) {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/index.js:31]:
```suggestion
	if (page === pages.COMMERCE && project && sessionId) {
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
@iagochaves the changes are good, but this event listener that is not removed, could be a problem for Application Performance

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js (hunk 2 above)
Comment (iagochaves) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/context/index.js:55]: I'll be adding that! Thanks

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Thanks

**Comment by kevenleone** [comment]
ci:forward

**Comment by aildsonf** [comment]
Just starting reviewing :)

**Comment by eliasantosf** [comment]
Just starting reviewing :)

**Comment by claricebrasil** [comment]
Just starting reviewing :)

**Comment by claricebrasil** [comment]
Just starting reviewing :)

**Comment by joaosvassis** [comment]
Just starting reviewing :)

**Comment by iagochaves** [comment]
ci:test

**Comment by iagochaves** [comment]
ci:test:sf

**Comment by claricebrasil** [comment]
Just starting reviewing :)

**Comment by aildsonf** [comment]
Just starting reviewing :)

**Comment by aildsonf** [comment]
Just starting reviewing :)

**Comment by Abigailmelo** [comment]
Just starting reviewing :)

**Comment by iagochaves** [comment]
ci:forward

**Comment by iagochaves** [comment]
ci:close

**Comment by iagochaves** [comment]
re-sent https://github.com/pt-liferay-solutions/liferay-portal/pull/602

**Pull request closed without merging**